### PR TITLE
✨ TableReader end-to-end

### PR DESCRIPTION
**Pull request by RTLS**
Co-authored-by: Anthony Accomazzo <accomazz@gmail.com>

### CODE REVIEW

**Comment by RTLS** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/751?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#751** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/751?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/751?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#750** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/750?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#749** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/749?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#717** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/717?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#716** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/716?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#713** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/713?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#714](https://github.com/sequinstream/sequin/pull/714) <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/714?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: lib/sequin/databases_runtime/message_handler.ex [hunk 1]
```diff
@@ -2,11 +2,14 @@ defmodule Sequin.DatabasesRuntime.SlotProcessor.MessageHandler do
   @moduledoc false
```
Comment (acco) [inline lib/sequin/databases_runtime/message_handler.ex:2]: data structures look good, obviously not populating the primary keys yet etc

File: lib/sequin/databases_runtime/message_handler.ex [hunk 2]
```diff
@@ -127,8 +144,54 @@ defmodule Sequin.DatabasesRuntime.SlotProcessor.MessageHandler do
...
+          TableReaderServer.flush_batch(backfill_id, %{
+            batch_id: batch_id,
+            seq: batch.seq,
+            drop_pks: batch.primary_key_values
+          })
+
+        update_in(ctx.table_reader_batches, fn batches ->
+          Enum.reject(batches, &(&1.batch_id == batch_id))
+        end)
+    end
```
Comment (RTLS) [inline lib/sequin/databases_runtime/message_handler.ex:188]: Use Enum.filter and assert exactly one batch exists for the batch_id

File: lib/sequin/databases_runtime/table_reader_server.ex [hunk 3]
```diff
@@ -194,32 +202,66 @@ defmodule Sequin.DatabasesRuntime.TableReaderServer do
...
+        # Process pending flush immediately
+        state = %{
+          state_without_flush
+          | cursor: next_cursor,
+            task_ref: nil,
+            successive_failure_count: 0,
+            current_batch_id: nil
+        }
+
+        {state, _from} = flush_batch(state.current_batch_id, pending_flush, batch_data, state, nil)
```
Comment (RTLS) [inline lib/sequin/databases_runtime/table_reader_server.ex:233]: reverse order?

**Pull request merged**